testplan/testing/multitest/entries/assertions.py

```diff
@@ -645,19 +649,66 @@ def __init__(
         )
 
     def evaluate(self):
-        self.delta = list(
-            difflib.diff(
-                self.first,
-                self.second,
-                ignore_space_change=self.ignore_space_change,
-                ignore_whitespaces=self.ignore_whitespaces,
-                ignore_blank_lines=self.ignore_blank_lines,
-                unified=self.unified,
-                context=self.context,
-            )
-        )
+        if sys.platform != "win32":
+            self.delta = self._diff_process().splitlines(True)
+        else:
+            self.delta = list(self._diff_difflib())
         return self.delta == []
 
+    def _diff_difflib(self):
+        out = difflib.diff(
+            self.first,
+            self.second,
+            ignore_space_change=self.ignore_space_change,
+            ignore_whitespaces=self.ignore_whitespaces,
+            ignore_blank_lines=self.ignore_blank_lines,
+            unified=self.unified,
+            context=self.context,
+        )
+        return out
+
+    def _diff_process(self):
+        first = "".join(self.first)
+        second = "".join(self.second)
+        with tempfile.NamedTemporaryFile(
+            delete=False,
+            mode="w",
+        ) as first_file:
+            first_file.write(first)
+        with tempfile.NamedTemporaryFile(
+            delete=False,
+            mode="w",
+        ) as second_file:
+            second_file.write(second)
+
+        cmd = ["diff"]
```

Comment:
shall we specify full path to diff binary, e.g /usr/bin/diff?

Reply:
Making the change.